nncf/torch/quantization/quantize.py

```diff
@@ -0,0 +1,191 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from copy import deepcopy
+from typing import Any, Dict, Optional, Tuple
+
+import torch
+from nncf.common.quantization.structs import QuantizationPreset
+from nncf.config import NNCFConfig
+from nncf.config.structures import BNAdaptationInitArgs
+from nncf.config.structures import QuantizationRangeInitArgs
+from nncf.data import Dataset
+from nncf.parameters import ModelType
+from nncf.parameters import TargetDevice
+from nncf.torch.dynamic_graph.io_handling import wrap_nncf_model_inputs_with_objwalk
+from nncf.torch.dynamic_graph.io_handling import wrap_nncf_model_outputs_with_objwalk
+from nncf.torch.initialization import PTInitializingDataLoader
+from nncf.torch.model_creation import create_compressed_model
+from nncf.torch.nested_objects_traversal import objwalk
+from nncf.torch.utils import get_model_device
+from nncf.torch.utils import is_tensor
+
+
+# TODO(alexsu52): It is a workaround and should be removed.
```

Comment:
Same - elaborate when and how it should be removed.

Reply:
Thanks for the comment. I filed the task ref: 95788